core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -667,7 +675,10 @@ private Expression translate0(RexNode expr, RexImpTable.NullAs nullAs,
       }
       InputGetter getter =
           correlates.apply(((RexCorrelVariable) target).getName());
-      return getter.field(list, fieldAccess.getField().getIndex(), storageType);
+      Expression y = getter.field(list, fieldIndex, storageType);
+      Expression input = list.append("corInp" + fieldIndex + "_", y); // safe to share
```

Comment:
This is not required.

Does it really add any value?

Reply:
I have it like this for two reasons:
- consistency with the existing code 639-641 (case INPUT_REF);
- readability of generated code.